test/200_fullchain-bot.test.ts

```diff
@@ -0,0 +1,314 @@
+        ({
+            requester,
+            appProvider,
+            datasetProvider,
+            scheduler,
+            anyone,
+            worker1,
+            worker2,
+            worker3,
+            worker4,
+            worker5,
+        } = accounts);
+        iexecWrapper = new IexecWrapper(proxyAddress, accounts);
+        ({ appAddress, datasetAddress, workerpoolAddress } = await iexecWrapper.createAssets());
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+        ordersActors = {
+            appOwner: appProvider,
+            datasetOwner: datasetProvider,
+            workerpoolOwner: scheduler,
+            requester: requester,
+        };
+        ordersAssets = {
+            app: appAddress,
+            dataset: datasetAddress,
+            workerpool: workerpoolAddress,
+        };
+        ordersPrices = {
+            app: appPrice,
+            dataset: datasetPrice,
+            workerpool: workerpoolPrice,
+        };
+    }
+
+    it('Task Lifecycle with BoT Replication and Error Handling', async function () {
+        const workers = [worker1, worker2, worker3, worker4, worker5];
+        // Create deal.
+        const volume = 3;
+        const tasksAndWorkers: {
+            [key: number]: {
+                worker: SignerWithAddress;
+                useEnclave: boolean;
+                result: string;
+            }[];
+        } = {
+            0: [
+                { worker: worker1, useEnclave: false, result: 'iExec BOT 0' },
+                { worker: worker2, useEnclave: false, result: 'iExec BOT 0' },
+            ],
+            1: [
+                { worker: worker2, useEnclave: true, result: 'iExec BOT 1' },
+                { worker: worker3, useEnclave: true, result: 'iExec BOT 1' },
+            ],
+            2: [
+                { worker: worker1, useEnclave: false, result: 'iExec BOT 2' },
+                { worker: worker3, useEnclave: false, result: '<timeout reached>' },
+                { worker: worker2, useEnclave: true, result: 'iExec BOT 2' },
+                { worker: worker4, useEnclave: true, result: 'iExec BOT 2' },
+                { worker: worker5, useEnclave: true, result: 'iExec BOT 2' },
+            ],
+        };
+        const orders = buildOrders({
+            assets: ordersAssets,
+            prices: ordersPrices,
+            requester: requester.address,
+            tag: standardDealTag,
+            volume,
+            trust: 4,
+        });
+        const { dealId, dealPrice, schedulerStakePerDeal } = await iexecWrapper.signAndMatchOrders(
+            ...orders.toArray(),
+        );
+        const taskPrice = appPrice + datasetPrice + workerpoolPrice;
+        const schedulerStakePerTask = schedulerStakePerDeal / volume;
+        const workerRewardPerTask = await iexecWrapper.computeWorkerRewardPerTask(
+            dealId,
+            PocoMode.CLASSIC,
+        );
```

Comment:
```suggestion
        const workersRewardPerTask = await iexecWrapper.computeWorkersRewardPerTask(
            dealId,
            PocoMode.CLASSIC,
        );
```
We probably need to rename it in other files.

Reply:
Will rename it in the next PR that will concern the 000 fullchain test file thanks